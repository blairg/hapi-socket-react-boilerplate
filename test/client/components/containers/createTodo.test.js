/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/extensions */

import React from 'react';
import Axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import CreateTodo from './../../../../src/client/components/containers/createTodo.jsx';

const middlewares = [];
const mockStore = configureStore(middlewares);
const todo = { title: 'my title', body: 'my body' };
const initialState = { setTodos: { todos: [todo] } };
const store = mockStore(initialState);

let mockAxios;

describe('client/components/containers/createTodo -> <CreateTodo />', () => {
  const event = { preventDefault: () => {} };

  beforeEach(() => {
    mockAxios = new MockAdapter(Axios);
    jest.spyOn(event, 'preventDefault');
  });

  afterEach(() => {
    mockAxios.reset();
  });

  describe('CreateTodo.handleDelete(event)', () => {
    test('should call Axios.delete and event.preventDefault()', async () => {
      mockAxios.onDelete('/todos').reply(200);
      const success = await CreateTodo.handleDelete(event);

      expect(event.preventDefault).toBeCalled();
      expect(success).toBeTruthy();
    });

    test('should call Axios.delete and throw error', async () => {
      mockAxios.onDelete('/todos').networkError();
      const success = await CreateTodo.handleDelete(event);

      expect(event.preventDefault).toBeCalled();
      expect(success).toBeFalsy();
    });
  });

  describe('<CreateTodo />', () => {
    test('should render correctly', () => {
      const tree = renderer.create(<CreateTodo store={store} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
