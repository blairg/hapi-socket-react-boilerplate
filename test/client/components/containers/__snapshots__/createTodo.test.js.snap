// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/components/containers/createTodo -> <CreateTodo /> <CreateTodo /> should render correctly 1`] = `
<div
  className="container"
  id="blogContainer"
>
  <div
    className="row"
  >
    <div
      className="six columns blogEntriesCreator"
    >
      <h1>
        Create a Todo
      </h1>
      <div
        className="row form-field"
      >
        <label
          className="two columns label--required"
          htmlFor="title"
        >
          Title
        </label>
        <section>
          <input
            className="ten columns"
            id="title"
            name="title"
            onChange={[Function]}
            required={true}
            type="text"
            value={undefined}
          />
        </section>
      </div>
      <div
        className="row form-field"
      >
        <label
          className="two columns label--required"
          htmlFor="body"
        >
          Body
        </label>
        <section>
          <input
            className="ten columns"
            id="body"
            name="body"
            onChange={[Function]}
            required={true}
            type="text"
            value={undefined}
          />
        </section>
      </div>
      <div
        className="row form-buttons"
      >
        <button
          className="six columns delete-button"
          onClick={[Function]}
          type="submit"
        >
          Delete All
        </button>
        <button
          className="six columns button-primary"
          onClick={[Function]}
          type="submit"
        >
          Create
        </button>
      </div>
    </div>
    <div
      className="six columns blogEntriesContainer"
    >
      <div
        className="blogEntries"
        id="blogEntries"
      />
    </div>
  </div>
</div>
`;
